R/R/prophet.R

```diff
@@ -1274,9 +1288,134 @@ fit.prophet <- function(m, df, ...) {
 #' forecast <- predict(m, future)
 #' plot(m, forecast)
 #' }
-#'
 #' @export
 predict.prophet <- function(object, df = NULL, ...) {
+  
+  if (is.null(df)) {
+    m <- object
+    history.and.future <- m$history
+  } else {
+    if (nrow(df) == 0) {
+      stop("Dataframe has no rows.")
+    }
+    out <- setup_dataframe(object, df)
+    history.and.future <- out$df
+    m <- out$m
+  }
+  
+  out2 <- make_all_seasonality_features(m, history.and.future)
+  seasonal.features <- out2$seasonal.features
+  component.cols <- object$train.component.cols
+  
+  # Construct input to stan
+  dat <- list(
+    T = nrow(object$history),
+    K = ncol(seasonal.features),
+    S = length(object$changepoints.t),
+    y = object$history$y_scaled,                         
+    t = object$history$t,                                
+    cap = rep(0, nrow(object$history)),                         
+    t_change = array(object$changepoints.t),
+    X = as.matrix(seasonal.features[1:nrow(object$history),]),            
+    sigmas = array(out2$prior.scales),
+    tau = object$changepoint.prior.scale,
+    trend_indicator = as.numeric(object$growth == 'logistic'),
+    s_a = array(component.cols$additive_terms),
+    s_m = array(component.cols$multiplicative_terms),
+    
+    T_pred = nrow(history.and.future),
+    t_pred = history.and.future$t,
+    cap_pred = 
+      if(class(history.and.future$cap) == "NULL") { 
```

Comment:
My personal preference is `'cap' %in% colnames(history.and.future)` since I think it reads a little more cleanly (and more closely parallels the python syntax)